pagopa/pn-bff
# Mandate api migration - pn-10952

## Short description

Migrated mandate api.

## List of changes proposed in this pull request

- Added new openapi definition file
- Updated pom with two new execution tasks to automatically generate clients
- Added mandate client, service and controller
- Added mappers to transform data from mandate ms to bff and viceversa
- Added tests

## How to test

Check that the following api work
- GET /bff/v1/mandate/delegate/count with PG user
- POST /bff/v1/mandate with PF/PG
- PATCH /bff/v1/mandate/{mandateId}/accept with PF/PG
- PATCH /bff/v1/mandate/{mandateId}/revoke with PF/PG
- PATCH /bff/v1/mandate/{mandateId}/reject with PF/PG
- PATCH /bff/v1/mandate/{mandateId}/update with PG
- GET /bff/v1/mandate/delegate with PF
- POST /bff/v1/mandate/delegate with PG
- GET /bff/v1/mandate/delegator with PF/PG
